build-tools/packages/build-tools/src/common/biome2Config.ts

```diff
@@ -0,0 +1,420 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert/strict";
+import { stat } from "node:fs/promises";
+import path from "node:path";
+import { merge } from "ts-deepmerge";
+import type { Opaque } from "type-fest";
+
+import type { Configuration as Biome2ConfigRaw } from "./biome2ConfigTypes";
+import {
+	filterFilesWithPatterns,
+	loadRawBiomeConfigFile,
+	resolveExtendsChainGeneric,
+} from "./biomeConfigUtils";
+import type { GitRepo } from "./gitRepo";
+
+// switch to regular import once building ESM
+const findUp = import("find-up");
+
+/**
+ * Convenience type to represent a Biome 2.x config that has been loaded while following and merging the
+ * "extends" values. This helps differentiate between the single loaded configs and the fully resolved config.
+ */
+export type Biome2ConfigResolved = Opaque<Biome2ConfigRaw, "Biome2ConfigResolved">;
+
+/**
+ * Loads a Biome 2.x configuration file _without_ following any 'extends' values. You probably want to use
+ * {@link loadBiome2Configs} instead of this function.
+ */
+async function loadRawBiome2Config(configPath: string): Promise<Biome2ConfigRaw> {
+	return loadRawBiomeConfigFile<Biome2ConfigRaw>(configPath);
+}
```

Comment:
SO much of this file seems like it could be reused from v1. For each function, explain why there MUST be two versions or why that is better than sharing or creating a generic version.